concent_api/core/payments/backends/sci_backend.py

```diff
@@ -55,6 +57,46 @@ def get_list_of_payments(
     return payments_list
 
 
+def are_all_payment_ts_younger_than_last_payment_closure_time_if_payment_exists(
+    requestor_eth_address: str,
+    provider_eth_address: str,
+    subtask_results_accepted_list: List[ForcedPaymentEvent],
+) -> bool:
+
+    oldest_payments_ts = _get_oldest_payment_timestamp_from_subtask_results_accepted_list(subtask_results_accepted_list)
+
+    forced_payment_event_list = _get_list_of_forced_payment_events(
+        requestor_eth_address=requestor_eth_address,
+        provider_eth_address=provider_eth_address,
+        search_payments_since_ts=oldest_payments_ts,
+    )
+
+    for forced_payment_event in forced_payment_event_list:
```

Comment:
I don't see a bug here. The code fetches all settlement payments with block timestamp later or equal to `oldest_payments_ts` which means all the ones with `closure_time >= oldest_payments_ts` (the ones we're actually interested in) and possibly some with earlier `closure_time`. We can be sure that all the existing settlement payments with `closure_time >= oldest_payments_ts` are on our list because the contract validates that `closure_time` <= block timestamp. Then the loop goes over all the events and checks if any of them actually has `closure_time >= oldest_payments_ts`. If so, returns `False`.

Note that if `closure_time ==` block timestamp was always true, we wouldn't even need a loop - we could just check if the list is empty or not. Unfortunately that's not the case the `closure_time` does not have to match block timestamp.

BTW, @pawelkisielewicz it's called `payment_ts` not `payments_ts` in golem-messages. Please use that name for consistency.

Reply:
renamed